### PR TITLE
ticket56 - add ProductCounter

### PR DESCRIPTION
**Pull request by JimHofman**

### CODE REVIEW

**Comment by JimHofman** [comment]
Thought this should be an easy merge, and made sense to merge as we go.

File: src/opera/pge/dswx_pge.py [hunk 1]
```diff
@@ -180,20 +180,21 @@ def _core_filename(self, inter_filename=None):
         dataset_fields = get_hls_filename_fields(f"{dataset}.BAND.tif")
```
Comment (collinss-jpl) [inline src/opera/pge/dswx_pge.py:180]: Docstring should be updated

File: src/opera/pge/dswx_pge.py [hunk 2]
```diff
@@ -180,20 +180,21 @@ def _core_filename(self, inter_filename=None):
...
         tile_id = dataset_fields['tile_id']
         timetag = dataset_fields['acquisition_time']
         version = dataset_fields['collection_version']
         subversion = dataset_fields['sub_version']
 
         # Assign the core file to the cached class attribute
         self._cached_core_filename = (
-            f"{self.PROJECT}_{self.LEVEL}_{self.NAME}_{source}_{sensor}_{tile_id}_{timetag}_{version}.{subversion}"
+            f"{self.PROJECT}_{self.LEVEL}_{self.NAME}_{source}_{spacecraft_name}_"
+            f"{tile_id}_{timetag}_{str(self.runconfig.product_counter).zfill(3)}_{version}.{subversion}"
```
Comment (collinss-jpl) [inline src/opera/pge/dswx_pge.py:192]: For consistency with NISAR, lets put the product counter at the very end, just prior to the file extension.

File: src/opera/pge/dswx_pge.py [hunk 3]
```diff
@@ -180,20 +180,21 @@ def _core_filename(self, inter_filename=None):
...
         )
 
         return self._cached_core_filename
 
-    def _geotiff_filename(self, inter_filename):
+    def _geotiff_filename(self, inter_filename=None):
```
Comment (collinss-jpl) [inline src/opera/pge/dswx_pge.py:197]: Investigate if this is still needed

**Review by collinss-jpl — CHANGES_REQUESTED** [review]

File: src/opera/test/pge/test_dswx_pge.py [hunk 4]
```diff
@@ -294,6 +295,23 @@ def test_dswx_pge_output_validation(self):
             if os.path.exists(test_runconfig_path):
                 os.unlink(test_runconfig_path)
 
+    def test_geotiff_filename(self):
```
Comment (collinss-jpl) [inline src/opera/test/pge/test_dswx_pge.py:298]: This unit test needs to be patched by copying this decorator:

 @patch.object(opera.util.img_utils, "get_geotiff_metadata", get_geotiff_metadata_patch)

File: src/opera/pge/dswx_pge.py (hunk 3 above)
Comment (JimHofman) [reply src/opera/pge/dswx_pge.py:197]: After making all the other changes I reran the unit test with the original line and it worked fine, so I removed '=None'

File: src/opera/pge/dswx_pge.py (hunk 1 above)
Comment (JimHofman) [reply src/opera/pge/dswx_pge.py:180]: Completed

File: src/opera/pge/dswx_pge.py (hunk 2 above)
Comment (JimHofman) [reply src/opera/pge/dswx_pge.py:192]: completed

File: src/opera/test/pge/test_dswx_pge.py (hunk 4 above)
Comment (JimHofman) [reply src/opera/test/pge/test_dswx_pge.py:298]: completed

Comment (collinss-jpl) [reply src/opera/test/pge/test_dswx_pge.py:299]: Just FYI, there was one more minor change needed to make this test work within the container: the test itself was also calling the unpatched version of `get_geotiff_metadata()`, it now calls the patched version manually so the test is consistent both in and out of a container.

**Review by collinss-jpl — APPROVED** [review]

**Pull request merged**